plugins/modules/win_scoop.ps1

```diff
@@ -0,0 +1,230 @@
+
+  $scoop_app = Get-Command -Name scoop.ps1 -Type ExternalScript -ErrorAction SilentlyContinue
+  if ($null -eq $scoop_app) {
+    # We need to install scoop
+    # Enable TLS1.1/TLS1.2 if they're available but disabled (eg. .NET 4.5)
+    $security_protocols = [Net.ServicePointManager]::SecurityProtocol -bor [Net.SecurityProtocolType]::SystemDefault
+    if ([Net.SecurityProtocolType].GetMember("Tls11").Count -gt 0) {
+      $security_protocols = $security_protcols -bor [Net.SecurityProtocolType]::Tls11
+    }
+    if ([Net.SecurityProtocolType].GetMember("Tls12").Count -gt 0) {
+      $security_protocols = $security_protcols -bor [Net.SecurityProtocolType]::Tls12
+    }
+    [Net.ServicePointManager]::SecurityProtocol = $security_protocols
+
+    $client = New-Object -TypeName System.Net.WebClient
+
+    $script_url = "https://get.scoop.sh"
+
+    try {
+      $install_script = $client.DownloadString($script_url)
+    }
+    catch {
+      $module.FailJson("Failed to download Scoop script from '$script_url'; $($_.Exception.Message)", $_)
+    }
+
+    if (-not $module.CheckMode) {
+      $res = Run-Command -Command "powershell.exe -" -stdin $install_script -environment $environment
+      if ($res.rc -ne 0) {
+        $module.Result.rc = $res.rc
+        $module.Result.stdout = $res.stdout
+        $module.Result.stderr = $res.stderr
+        $module.FailJson("Scoop bootstrap installation failed.")
+      }
+      $module.Warn("Scoop was missing from this system, so it was installed during this task run.")
+    }
+    $module.Result.changed = $true
+
+    # Refresh PATH
+    $env:Path = [System.Environment]::GetEnvironmentVariable("Path", "Machine") + ";" + [System.Environment]::GetEnvironmentVariable("Path", "User")
+
+    # locate the newly installed scoop.ps1
+    $scoop_app = Get-Command -Name scoop.ps1 -Type ExternalScript -ErrorAction SilentlyContinue
+  }
+
+  if ($module.CheckMode -and $null -eq $scoop_app) {
+    $module.Result.skipped = $true
+    $module.Result.msg = "Skipped check mode run on win_scoop as scoop.ps1 cannot be found on the system"
+    $module.ExitJson()
+  }
+
+  if ($null -eq $scoop_app -or -not (Test-Path -LiteralPath $scoop_app.Path)) {
+    $module.FailJson("Failed to find scoop.ps1, make sure it is added to the PATH")
+  }
+
+  return $scoop_app.Path
+}
+
+function Get-ScoopPackages {
+  param(
+    [Parameter(Mandatory = $true)] [string]$scoop_path
+  )
+
+  $command = Argv-ToString -arguments @("powershell.exe", $scoop_path, "export")
+  $res = Run-Command -Command $command
+  if ($res.rc -ne 0) {
+    $module.Result.command = $command
+    $module.Result.rc = $res.rc
+    $module.Result.stdout = $res.stdout
+    $module.Result.stderr = $res.stderr
+    $module.FailJson("Error checking installed packages")
+  }
+
+  $installed_packages = $res.stdout -split "`n" |
+  Select-String '(.*?) \(v:(.*?)\) \[(.*?)\]' |
```

Comment:
This is a nit as it's your module but I find it's nice to at least indent the pipeline operations compared to the original call so we can see that this is running with pipeline input rather than a direct invocation of the cmdlet.

```powershell
$installed_packages = $res.stdout -split "`n" |
  Select-String '(.*?) \(v:(.*?)\) \[(.*?)\]' |
  ForEach-Object {
    ...
  }
```

Reply:
I'd be fine with this change on it's own, but with the below change about omitting the explicit return value, I think it looks better without over-indenting the first line.

Reply:
The trouble that I find with this form is that it makes it look like `Select-String` and `ForEach-Object` their own call without any input going into it. I had to double take why the `ForEach-Object` was working without the `($var in $coll)` part but then realised it was the `ForEach-Object` cmdlet instead.